src/std/main.ab

```diff
@@ -239,3 +239,39 @@ pub fun get_env_var(var: Text): Text {
 pub fun load_env_file(): Null {
     unsafe $export "\$(xargs < .env)" > /dev/null$
 }
+
+pub fun shell_isset(name: Text): Bool {
+    $[[ ! -z \$\{!name+z} ]]$ failed {
+        return false
+    }
+    return true
+}
+
+pub fun shell_constant_set(name: Text, val: Text): Bool {
+    $readonly \$name="\$val" 2> /dev/null$ failed {
```

Comment:
There is no guarantee that variable "val" will be called "val" after translation. It could be called differently with some numbers or tags like "val_0" or whatever this is why if you want to get the mangled name of this variable, please use `nameof val` instead:
```suggestion
    $readonly \${nameof name}="\${nameof val}" 2> /dev/null$ failed {
```